packages/eslint-plugin/src/rules/unbound-method.ts

```diff
@@ -197,6 +197,16 @@ function isDangerousMethod(symbol: ts.Symbol, ignoreStatic: boolean): boolean {
     return false;
   }
 
+  if (valueDeclaration.kind === ts.SyntaxKind.PropertyDeclaration) {
+    const propertyDeclaration = valueDeclaration as ts.PropertyDeclaration;
+    if (
+      propertyDeclaration.initializer &&
+      propertyDeclaration.initializer.kind === ts.SyntaxKind.FunctionExpression
+    ) {
+      return true;
+    }
+  }
```

Comment:
nit - move this into the switch below eg

```ts
case ts.SyntaxKind.PropertyDeclaration:
  const propertyDeclaration = valueDeclaration as ts.PropertyDeclaration;
  return (
    propertyDeclaration.initializer?.kind === ts.SyntaxKind.FunctionExpression
  );
```

Reply:
oof, nice catch!